### PR TITLE
Reduce server service dependencies

### PR DESCRIPTION
**Pull request by krusche**
Another attempt to reduce the dependencies between services on the server side

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.

### Steps for Testing
only code review is needed

### CODE REVIEW

**Review by PatZintel — DISMISSED** [review]
Seems fine to me, only did code review

**Review by joschahenningsen — APPROVED** [review]
Mostly did some integration testing which went well. IMO this makes things quite a bit cleaner :+1:

File: src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java [hunk 1]
```diff
@@ -4,10 +4,7 @@
 
 import java.time.ZonedDateTime;
 import java.time.temporal.ChronoUnit;
-import java.util.Collection;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java:7]: I think those things are handled by Spotless / the IntelliJ IDE

File: src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java [hunk 2]
```diff
@@ -1,13 +1,30 @@
 package de.tum.in.www1.artemis.repository;
 
+import static de.tum.in.www1.artemis.config.Constants.FEEDBACK_DETAIL_TEXT_MAX_CHARACTERS;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.ArrayList;
```
[the remark this answers is not in the document]
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java:7]: I think those things are handled by Spotless / the IntelliJ IDE

**Pull request merged**